src/rudder_wrapper.rs

```diff
@@ -69,11 +81,18 @@ impl RudderWrapper {
         let anonymous_id = self.get_anonymous_id();
         
         let user_id = {self.config.lock().unwrap().user_id().map(|id| id.to_string())};
+        let os  = {self.config.lock().unwrap().get_os().map(|os| os.to_string())};
```

Comment:
@cmgreen210 just small comment on this. if possible try to only lock the mutex once. just have a block where you lock it get all the values you need then return them.